### DISCUSSION TITLE
.NET Enumeration Make it configurable to add spacing between words?

### DISCUSSION THREAD

**Discussion by erikjanwestendorp** [leekelleher/umbraco-contentment#228] [Question] [answered]
I really like the DataList with a .NET Enumeration as DataSource. 😄. 

Is there a way to stop adding spacing between words? In most cases it is really useful and inventive, but in some cases it is not. For example, if we are talking about social media. See the image below. 

![image](https://user-images.githubusercontent.com/7831614/173189972-3c656239-709b-4fae-b597-d1533e0487ae.png)

**Comment by leekelleher** [comment, marked as answer]
@erikjanwestendorp Do you have control over the enum code? If so, you could use the [`DataListItemAttribute`](https://github.com/leekelleher/umbraco-contentment/blob/3.3.1/src/Umbraco.Community.Contentment/DataEditors/DataList/DataListItemAttribute.cs) to annotate the field.

Similar to how the enum I have on the demo site: https://github.com/leekelleher/umbraco-contentment/blob/3.3.1/src/Umbraco.Cms.9.x/DataSources/RenderPosition.cs#L7

So for your example, you'd have something like...

```csharp


public enum SocialMedia
{
	Facebook,

	[DataListItem(Name = "LinkedIn")]
	LinkedIn,

	Twitter,

	[DataListItem(Name = "WhatsApp")]
	WhatsApp,
}
```

Would that work for you?

**Comment by erikjanwestendorp** [comment]
@leekelleher Yes I have control over the enum code and annotate the field with the `DataListItem` attribute works great.  Thanks! 😄
